egg/zoo/compo_vs_generalization/train.py

```diff
@@ -201,32 +232,57 @@ def main(params):
     device = opts.device
     print(opts)
 
-    full_data = enumerate_attribute_value(opts.n_attributes, opts.n_values)
-    if opts.density_data > 0:
-        sampled_data = select_subset_V2(
-            full_data, opts.density_data, opts.n_attributes, opts.n_values
+    if opts.build_full_dataset:
+        print("WARNING, using deprecated code")
+        full_data = enumerate_attribute_value(opts.n_attributes, opts.n_values)
+        if opts.density_data > 0:
+            sampled_data = select_subset_V2(
+                full_data, opts.density_data, opts.n_attributes, opts.n_values
+            )
+            full_data = copy.deepcopy(sampled_data)
+
+        train, generalization_holdout = split_holdout(full_data)
+        train, uniform_holdout = split_train_test(train, 0.1)
+
+        generalization_holdout, train, uniform_holdout, full_data = [
+            one_hotify(x, opts.n_attributes, opts.n_values)
+            for x in [generalization_holdout, train, uniform_holdout, full_data]
+        ]
+
+        train, validation = ScaledDataset(train, opts.data_scaler), ScaledDataset(train, 1)
+
+        generalization_holdout, uniform_holdout, full_data = (
+            ScaledDataset(generalization_holdout),
+            ScaledDataset(uniform_holdout),
+            ScaledDataset(full_data),
         )
-        full_data = copy.deepcopy(sampled_data)
+        generalization_holdout_loader, uniform_holdout_loader, full_data_loader = [
+            DataLoader(x, batch_size=opts.batch_size)
+            for x in [generalization_holdout, uniform_holdout, full_data]
+        ]
 
-    train, generalization_holdout = split_holdout(full_data)
-    train, uniform_holdout = split_train_test(train, 0.1)
+    else:
+        rng = torch.Generator()
+        rng.manual_seed(opts.data_seed)
+        train, test, validation = \
+            build_datasets(opts.n_attributes, 
+                           opts.n_values,
+                           opts.train_size,
+                           opts.test_size,
+                           opts.validation_size,
+                           rng=rng)
 
-    generalization_holdout, train, uniform_holdout, full_data = [
-        one_hotify(x, opts.n_attributes, opts.n_values)
-        for x in [generalization_holdout, train, uniform_holdout, full_data]
-    ]
+        train, validation, test = [
+            one_hotify(x, opts.n_attributes, opts.n_values)
+            for x in [train, validation, test]
+        ]
 
-    train, validation = ScaledDataset(train, opts.data_scaler), ScaledDataset(train, 1)
+        train = ScaledDataset(train, opts.data_scaler)
+        validation = ScaledDataset(validation, 1)
+        test = ScaledDataset(test)
```

Comment:
You probably don't need this wrapper here since we're not scaling the test and validation data